subsys/bluetooth/host/gatt.c

```diff
@@ -5204,19 +5211,25 @@ void bt_gatt_cancel(struct bt_conn *conn, void *params)
 	}
 }
 
+#if defined(CONFIG_BT_GATT_AUTO_RESUBSCRIBE)
 static void add_subscriptions(struct bt_conn *conn)
 {
 	struct gatt_sub *sub;
 	struct bt_gatt_subscribe_params *params;
 
+	if (!bt_addr_le_is_bonded(conn->id, &conn->le.dst)) {
+		return;
+	}
+
 	sub = gatt_sub_find(conn);
 	if (!sub) {
 		return;
 	}
 
 	/* Lookup existing subscriptions */
 	SYS_SLIST_FOR_EACH_CONTAINER(&sub->list, params, node) {
-		if (bt_addr_le_is_bonded(conn->id, &conn->le.dst) &&
+		if (!atomic_test_bit(params->flags,
+				     BT_GATT_SUBSCRIBE_FLAG_SENT) &&
```

Comment:
The `BT_GATT_SUBSCRIBE_FLAG_SENT` is ever only tested for bonded devices, but is unconditionally set and cleared. Is that intentional?

Reply:
yes, I figured adding another conditional when setting and clearing it wasn't necessary. I can add it of course, but I think the flag's comment is clear on how it should be used.

Reply:
Yeah, it's not really a problem :)